lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -1016,6 +1030,20 @@ static Optional<ObjCReason> shouldMarkClassAsObjC(const ClassDecl *CD) {
         .fixItRemove(attr->getRangeWithAt());
     }
 
+    // If the class has resilient ancestry, @objc just controls the runtime
+    // name unless -enable-resilient-objc-class-stubs is enabled.
+    if (ancestry.contains(AncestryFlags::HasResilientOther) &&
+        !ctx.LangOpts.EnableObjCResilientClassStubs) {
+      if (attr->hasName()) {
+        const_cast<ClassDecl *>(CD)->getAttrs().add(
+          new (ctx) ObjCRuntimeNameAttr(*attr));
```

Comment:
Do we have a test for this behavior already?

Reply:
I'll try to add one